service-common/rest/src/main/java/io/helidon/servicecommon/rest/HelidonRestServiceSupport.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.helidon.servicecommon.rest;
+
+import java.util.Objects;
+import java.util.logging.Logger;
+
+import io.helidon.config.Config;
+import io.helidon.webserver.Routing;
+import io.helidon.webserver.Service;
+import io.helidon.webserver.cors.CorsEnabledServiceHelper;
+import io.helidon.webserver.cors.CrossOriginConfig;
+
+/**
+ * Common base implementation for {@linkplain Service service} support classes which involve REST endpoints.
+ * <p>
+ *     This base class takes care of some tasks common to many services, using config and other settings in the builder:
+ *     <ul>
+ *         <li>Setting up the endpoint path (web context) for the service, using settings in the builder and config.</li>
+ *         <li>Providing automatic CORS support (and the ability to control it via config).</li>
+ *     </ul>
+ *
+ * <p>
+ *     Concrete implementations must implement {@link #postConfigureEndpoint(Routing.Rules)} to do any service-specific routing.
+ *     See also the {@link Builder} information for possible additional overrides.
+ * </p>
+ *
+ */
+public abstract class HelidonRestServiceSupport implements Service {
+
+    private final String context;
+    private final CorsEnabledServiceHelper corsEnabledServiceHelper;
+    private final Logger logger;
+
+    protected HelidonRestServiceSupport(Logger logger, Builder<?, ?> builder, String serviceName) {
```

Comment:
Protected constructors of public classes should have javadoc.

Reply:
Fixed. Ideally the javadoc plug-in would flag this.